app/routers/books.py

```diff
@@ -20,32 +25,40 @@
 # Local application imports
 # -------------------------
 from config import settings
-
 from ..applogger import rslogger
 from ..crud import create_useinfo_entry
 from ..db import database as db
 from ..schemas import LogItem, LogItemIncoming
 
-#
+# Routing
+# =======
 # Setup the router object for the endpoints defined in this file.  These will
-# be connected to the main application in main.py
-#
+# be `connected <included routing>` to the main application in `../main.py`.
 router = APIRouter(
-    prefix="/books",  # shortcut so we don't have to repeat this part
-    tags=["books"],  # groups all logger tags together in the docs
+    # shortcut so we don't have to repeat this part
+    prefix="/books",
+    # groups all logger `tags <https://fastapi.tiangolo.com/tutorial/path-operation-configuration/#tags>`_ together in the docs.
+    tags=["books"],
 )
 
 
-# Static Asset renderers
-# https://fastapi.tiangolo.com/tutorial/static-files/?h=+staticfiles#use-staticfiles
-# https://fastapi.tiangolo.com/advanced/custom-response/?h=+filere#fileresponse
-# for paths like: /books/published/basecourse/_static/rest
+# Options for static asset renderers:
+#
+# - `StaticFiles <https://fastapi.tiangolo.com/tutorial/static-files/?h=+staticfiles#use-staticfiles>`_. However, this assumes the static routes are known *a priori*, in contrast to books (with their static assets) that are dynamically added and removed.
+# - Manually route static files, returning them using a `FileResponse <https://fastapi.tiangolo.com/advanced/custom-response/#fileresponse>`_. This is the approach taken.
+#
+# for paths like: ``/books/published/basecourse/_static/rest``.
 # If it is fast and efficient to handle it here it would be great.  We currently avoid
-# any static file contact with web2py and handle static files upstream with nginx directly
-# Note the use of the path type for filepath in the decoration.  If you don't use path it
-# seems to only get you the ``next`` part of the path /pre/vious/next/the/rest
+# any static file contact with web2py and handle static files upstream with nginx directly; therefore, this is useful only for testing/a non-production environment.
+# Note the use of the ``path``` type for filepath in the decoration.  If you don't use path it
+# seems to only get you the ``next`` part of the path ``/pre/vious/next/the/rest``.
+#
+# :index:`todo`: **Routes for draft (instructor-only) books.**
 @router.get("/published/{course:str}/_static/{filepath:path}")
 async def get_static(course: str, filepath: str):
+    # :index:`fixme`: **Find a way to avoid a hard-coded path.** This fix is needed for the route below as well.
+    #
+    # :index:`todo`: **Are there any security concerns?** Could a malicious user pass a "path" of ``../../../private_files`` and access anything on the server?
```

Comment:
The browser subtracts the ../ from the URL path before sending it to the server. So we get a 404 and the get_static endpoint is not even called.

We could -- just to be safe -- make sure to either reject a relative path or make sure that our file path resolves to something below whatever we define as our BOOK_LIBRARY

Reply:
Sounds good. I try to be a bit paranoid about potential security issues. I could also argue that there's no need to fix this, since a production server should never be serving files this way. But...I think a relative path check or something like that would be best. I'd suggest the equivalent to [safe_join](https://flask.palletsprojects.com/en/1.1.x/api/#flask.safe_join) -- there's a copy of the relevant code in web2py's `books.py` starting at line 344.

Reply:
Do you want me to add this?

Reply:
Sure.

While you are in code review mode...  One question I've been thinking about is whether we need the `db` parameter on the function in crud.py or whether we should just import `db` into crud and use it.?

Also BTW, everything defined as a setting in config.py can be set by an environment variable.  Where the environment variable is the upper case version of the string in the settings class.  If you want to add book_library or book_folder as a setting to act as the base directory for books that would be great.

Reply:
Also, style-wise - sqlalchemy can use the bitwise operators just like DAL for and and or OR we can use the functional equivalent `and_` and `or_` in sqlalchemy, but I think we should pick one and be consistent.

Reply:
I think importing `db` is fine.

I'll think about nice ways to specify a path.

I strongly prefer operators like DAL -- the `and_` and `or_` functions are quite awkward.

Reply:
Done re: safe_join and book_path.